### PR TITLE
feat(ethers-core/block): add gas_target and next_block_base_fee fns a…

### PR DESCRIPTION
**Pull request by meetmangukiya**
…ccording to EIP-1559

## Motivation

Target gas usage and next block's base fee can be useful helper methods.

## Solution

Implemented the same referring the EIP.

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

File: ethers-core/src/types/block.rs [hunk 1]
```diff
@@ -89,6 +89,49 @@ pub struct Block<TX> {
...
+
+    /// The next block's base fee, it is a function of parent block's base fee and gas usage.
+    /// Reference: https://eips.ethereum.org/EIPS/eip-1559
+    pub fn next_block_base_fee(&self) -> Option<U256> {
+        let target_usage = self.gas_target();
+        if let None = self.base_fee_per_gas {
+            return None
+        }
+
+        let base_fee_per_gas = self.base_fee_per_gas.unwrap();
```
Comment (mattsse) [inline ethers-core/src/types/block.rs:111]: use `?` operator
```suggestion
        let base_fee_per_gas = self.base_fee_per_gas?;
```

File: ethers-core/src/types/block.rs [hunk 2]
```diff
@@ -89,6 +89,49 @@ pub struct Block<TX> {
...
+
+        if self.gas_used > target_usage {
```
Comment (mattsse) [inline ethers-core/src/types/block.rs:113]: this if elsif else can optionally be rewritten with a `match self.gas_used.cmp(&target_usage)`

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit

File: ethers-core/src/types/block.rs (hunk 2 above)
Comment (meetmangukiya) [reply ethers-core/src/types/block.rs]: Is it supposed to be like:
```rs
match self.gas_used.cmp(&target_usage) {
    Ordering::Less => {...},
    Ordering::Greater => {...},
    Ordering::Equal => {...},
}
```

Comment (mattsse) [reply ethers-core/src/types/block.rs]: yes!
usually, clippy complains if that's an if elseif else

**Review by gakonst — APPROVED** [review]

**Pull request merged**